backend/EasySpeak.Core/EasySpeak.Core.BLL/Services/UserService.cs

```diff
@@ -123,18 +151,27 @@ public async Task<LessonDto> EnrollUserToLesson(long lessonId)
     {
         var userId = _authService.UserId;
 
-        var user = _context.Users.SingleOrDefault(u => u.Id == userId) ?? throw new ArgumentException($"Failed to find the user with id {userId}");
-        var lesson = _context.Lessons.SingleOrDefault(l => l.Id == lessonId) ?? throw new ArgumentException($"Failed to find the lesson with id {lessonId}");
+        var user = _context.Users.SingleOrDefault(u => u.Id == userId) ??
+                   throw new ArgumentException($"Failed to find the user with id {userId}");
+        var lesson = _context.Lessons.SingleOrDefault(l => l.Id == lessonId) ??
+                     throw new ArgumentException($"Failed to find the lesson with id {lessonId}");
 
         user.Lessons.Add(lesson);
 
         await _context.SaveChangesAsync();
 
-        void AfterMapAction(Lesson o, LessonDto dto) => dto.SubscribersCount = _context.Lessons
-            .Select(t => new { Id = t.Id, SbCount = t.Subscribers.Count })
-            .FirstOrDefault(l => l.Id == lessonId)!.SbCount;
+        void AfterMapAction(Lesson o, LessonDto dto)
+        {
+            dto.SubscribersCount = _context.Lessons
+                .Select(t => new {Id = t.Id, SbCount = t.Subscribers.Count})
+                .FirstOrDefault(l => l.Id == lessonId)!.SbCount;
+        }
 
-        return _mapper.Map<Lesson, LessonDto>(lesson, options => options.AfterMap(AfterMapAction));
+        var lessonDto = _mapper.Map<Lesson, LessonDto>(lesson, options => options.AfterMap(AfterMapAction));
+        
+        SendAddClassQuery(_authService.UserId, lessonDto);
```

Comment:
Isn't it better to make it after lesson is started? As it may be cancelled and this values won't make any sense...

Reply:
I guess it's not needed anymore